provenance.jq

```diff
@@ -0,0 +1,75 @@
+# input: "build" object with platform and image digest
+#   $github: "github" context; CONTAINS SENSITIVE INFORMATION (https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/accessing-contextual-information-about-workflow-runs#github-context)
+#   $runner: "runner" context; https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/accessing-contextual-information-about-workflow-runs#runner-context
+#   $digest: the OCI image digest for the just-built image (normally in .build.resolved.annotations["org.opencontainers.image.ref.name"] but only post-push/regeneration and we haven't pushed yet)
+#
+# output: in-toto provenance statement (https://slsa.dev/spec/v1.0/provenance)
+#   see also: https://github.com/actions/buildtypes/tree/main/workflow/v1
+def github_actions_provenance($github; $runner; $digest):
+	if $github.event_name != "workflow_dispatch" then error("error: '\($github.event_name)' is not a supported event type for provenance generation") else
+	($github.workflow_ref | ltrimstr($github.repository + "/") | split("@") | if length == 2 then . else error("parsing 'workflow_ref' failed: '\(.)'") end) as $workflowRefSplit
```

Comment:
If we stick with this or something like it (https://github.com/docker-library/meta-scripts/pull/76#issuecomment-2361475268), it should move down into the workflow block, something like this:

```jq
	workflow: (
		($github.workflow_ref | ...) as $workflowRefSplit # or $workflowRefParts? if we split it more 🤔  with the capturing regex approach I put in my TODO comment, it could be the basis for the object and not `as` at all, instead just `+ { the missing bits }` 👀
		| {
			...
		}
	),
```